sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs

```diff
@@ -477,5 +515,33 @@ internal async Task CheckAndUpdateCancellationStatusAsync()
                 }
             }
         }
+
+        private void SetFailureType(string exceptionMessage)
+        {
+            lock (_failureTypeLock)
+            {
+                if (_failureType == JobPartFailureType.None)
+                {
+                    foreach (string errorMessage in DataMovementConstants.ErrorCode.CannotOverwrite)
```

Comment:
Hmm I don't love the raw error string comparison here... but I'm not sure what else to suggest. Seems like it's a combination of Storage error codes, which are relatively stable I guess, and some OS error messages.

For Blob error codes I assume we have constants for those defined somewhere in some package? Possible to at least use those or would that create a circular dependency? For OS error codes, I'm guessing those could be different on different OSes so not sure how to handle that.

Sorry not a totally helpful comment, since I don't have an alternative to suggest. If this is the way, then this is the way :).

Reply:
Yeah I'm also not happy about this either.

Let me see if I can do like a
```
if (exception is Azure.RequestFailedException)
if (exception is System.IO.FileAlreadyExistsException)
// (and some how check the error code too...)
```

We actually do have the blob error codes stored away but they're internal to the generated blobs package, I think I can pull them in somehow? I have to probably do some shared code work.

Reply:
For the constant blob error code: https://github.com/Azure/azure-sdk-for-net/issues/35553